driver/src/main/java/org/neo4j/driver/Config.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.driver;
+
+import java.util.logging.Level;
+
+import org.neo4j.driver.internal.logging.JULogging;
+import org.neo4j.driver.internal.spi.Logging;
+
+/**
+ * A configuration class to config driver properties.
+ * <p>
+ * To create a config:
+ * <pre>
+ * {@code
+ * Config config = Config
+ *                  .builder()
+ *               ** .setProperties() **
+ *                  .build();
+ * }
+ * </pre>
+ */
+public class Config
+{
+
+    /** A config with all default settings */
+    public static final Config DEFAULT_CONFIG = Config.builder().build();
```

Comment:
FWIW, I'm +1 on all those suggestions.

On Thu, Jun 18, 2015 at 3:37 AM, Chris Leishman <notifications@github.com>
wrote:

> In driver/src/main/java/org/neo4j/driver/Config.java
> <https://github.com/neo4j/neo4j-java-driver/pull/24#discussion_r32708777>:
>
> > + * <p>
> > + * To create a config:
> > + * <pre>
> > + * {@code
> > + * Config config = Config
> > + *                  .builder()
> > + *               ** .setProperties() **
> > + *                  .build();
> > + * }
> > + * </pre>
> > + */
> > +public class Config
> > +{
> > +
> > +    /** A config with all default settings */
> > +    public static final Config DEFAULT_CONFIG = Config.builder().build();
>
> Here's some suggestions:
>
> For this specific case (the default), you could introduce a factory method
> for this case, something like Config.empty() or Config.default()? I'd
> also suggest using that method wherever it's needed, instead of a public
> static field.
>
> I'd suggest renaming Config#builder() to Config#build(), so that it's a
> verb -- indicating the start of a sentence like "build with blah, with
> blah, and with blah". Likewise, perhaps use withXX() vs setXX() (we do
> this in some other builders
> <https://github.com/neo4j/neo4j/blob/2.3/community/logging/src/main/java/org/neo4j/logging/FormattedLogProvider.java#L73>
> ).
>
> Finally, you might like to rename ConfigBuilder#build() to
> ConfigBuilder#toConfig(), just so that it reads something like
> Config.build().withLogging(...).withConnectionPoolSize(...).toConfig(),
> which fits the sentence structure "build with logging, and with pool size,
> and then give me the Config object".
>
> Just suggestions :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/neo4j/neo4j-java-driver/pull/24/files#r32708777>.
>